scripts/check_extension_sync.py

```
@@ -0,0 +1,60 @@
+
+
+def extract_recommended_extensions(data: Dict[str, Any]) -> Set[str]:
+    return set(data.get("recommendations", []))
+
+
+def extract_devcontainer_extensions(data: Dict[str, Any]) -> Set[str]:
+    try:
+        return set(
+            ext if isinstance(ext, str) else ext["id"]
+            for ext in data["customizations"]["vscode"]["extensions"]
+        )
```

Ruff (0.11.9)

Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)